programs/pkey/rsa_encrypt.c

```diff
@@ -71,6 +74,16 @@ int main( int argc, char *argv[] )
     const char *pers = "rsa_encrypt";
     mbedtls_mpi N, E;
 
+#if defined(MBEDTLS_PLATFORM_C)
+    if( ( return_val = mbedtls_platform_setup( &platform_ctx ) ) != 0 )
+    {
+        mbedtls_printf( " failed\n  ! mbedtls_platform_setup returned %d\n",
+                        return_val );
+        exit_val = MBEDTLS_EXIT_FAILURE;
+        goto exit;
```

Comment:
This should be changed to `return ret`. If you go to exit here, you will free uninitialized resources, which might lead to undefined behavior, depending on implementation.